pkg/validate/serialize.go

```diff
@@ -31,14 +54,93 @@ func Verify(yamlFileName string) error {
 // (1) the yaml file can not be converted to json.
 // (2) there is a problem while unmarshalling in go type.
 // Returns an object of type olm.ClusterServiceVersion.
-func unmarshal(rawYAML []byte) (*olm.ClusterServiceVersion, error) {
+func unmarshal(rawYAML []byte) (olm.ClusterServiceVersion, error) {
 
 	var csv olm.ClusterServiceVersion
 
-	if err := yaml.Unmarshal(rawYAML, &csv); err != nil {
-		return nil, fmt.Errorf("error parsing CSV list (JSON) : %s", err)
+	rawJson, err := yaml.YAMLToJSON(rawYAML)
+	if err != nil {
+		fmt.Printf("error parsing raw YAML to Json: %s", err)
+		return csv, err
+	}
+	if err := json.Unmarshal(rawJson, &csv); err != nil {
+		return csv, fmt.Errorf("error parsing CSV list (JSON) : %s", err)
+	}
+
+	return csv, nil
+}
+
+func hasEmptyField(val interface{}) []string {
+
+	fieldValue := reflect.ValueOf(val)
+
+	switch fieldValue.Kind() {
+	case reflect.Struct:
+		return checkStructHasEmptyFields(fieldValue, []string{})
+	default:
+		return []string{"Error: input file is not aligned with Operator-Lifecycle-Manager's CSV type."}
 	}
+}
+
+// Takes in a string slice and checks if a string (x) is present in the slice.
+// Return true if the string is present in the slicec.
+func contains(a []string, x string) bool {
+	for _, n := range a {
+		if strings.ToLower(x) == strings.ToLower(n) {
+			return true
+		}
+	}
+	return false
+}
+
+// Recursive function that traverses a nested struct passed in as reflect value, and reports for errors/warnings
+// in case of null struct field values.
+// Returns a log of errors as slice of strings.
+func checkStructHasEmptyFields(v reflect.Value, log []string) []string {
```

Comment:
This function name could be improved. I'd suggest something along the lines of `csvReport` or `csvInspect`

Reply:
How about I change `hasEmptyField` to `csvInspect` and `checkStructHasEmptyFields` to `getMissingFields` or `getEmptyFields` because `checkStructHasEmptyFields` has nothing to do with `csv` in general?

Reply:
Great point! I agree with `csvInspect` instead of `hasEmptyField` and `checkMissingFields` instead of `checkStructHasEmptyFields`

Reply:
Done.